src/components/AppComponent.tsx

```diff
@@ -144,6 +157,11 @@ const AppComponent: FC<Props> = props => {
     }
   }, [showSplitView])
 
+  useEffect(() => {
+    if (!authenticated && userStateRef.current) dispatch(resetSettings())
+    userStateRef.current = authenticated
+  }, [authenticated])
```

Comment:
`useEffect` is intended for side effects that need to occur when a component is mounted. Logging out when the authentication state changes is not related to `AppComponent`. You've hooked into the component's rendering life cycle in order to respond to a changing state. A better place for this kind of logic is in an action-creator. A thunk action-creator can update state and produce side effects in response to events upstream from components. In this case, go straight to the source: directly to Firebase in the [onAuthStateChange](https://github.com/cybersemics/em/blob/dev/src/initialize.ts#L57) handler.

Reply:
Hmm.. Sounds better than what I did. I will change this.